src/daemon.js

```diff
@@ -78,30 +78,38 @@ function parseConfig (path, callback) {
   ], callback)
 }
 
+function tempDir (isJs) {
+  return join(os.tmpdir(), `${isJs ? 'jsipfs' : 'ipfs'}_${String(Math.random()).substr(2)}`)
+}
+
 /**
  * Controll a go-ipfs node.
  */
 class Node {
   /**
    * Create a new node.
    *
-   * @param {string} path
    * @param {Object} [opts]
    * @param {Object} [opts.env={}] - Additional environment settings, passed to executing shell.
-   * @param {boolean} [disposable=false] - Should this be a temporary node.
    * @returns {Node}
    */
-  constructor (path, opts, disposable) {
-    this.path = path
-    this.opts = opts || {}
-    this.exec = process.env.IPFS_EXEC || ipfsDefaultPath
+  constructor (opts) {
+    const rootPath = process.env.testpath ? process.env.testpath : __dirname
+    const isJs = truthy(process.env.IPFS_JS)
```

Comment:
Weird, if the class says it is to control a gonode, why it is checking JS stuff?

Reply:
same as above, controls both, go and js nodes.